paimon-filesystems/paimon-oss-impl/src/main/java/org/apache/paimon/oss/OSSFileIO.java

```diff
@@ -124,7 +134,22 @@ protected FileSystem createFileSystem(org.apache.hadoop.fs.Path path) {
                         throw new UncheckedIOException(e);
                     }
                     return fs;
-                });
+                };
+
+        if (allowCache) {
+            return CACHE.computeIfAbsent(
+                    new CacheKey(hadoopOptions, scheme, authority), key -> supplier.get());
+        } else {
+            return supplier.get();
+        }
+    }
+
+    @Override
+    public void close() {
+        if (!allowCache) {
```

Comment:
This means, if allowCache, the fileIo need to close by outside?

Reply:
if not allow cache, the fileio should be closed.

If allow cache, the fileio is in the static cache, we cannot close them.